src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-gateway-group/nvmeof-gateway-group.component.ts

```diff
@@ -0,0 +1,124 @@
+import { catchError, map, switchMap } from 'rxjs/operators';
+import { GatewayGroup, NvmeofService } from '~/app/shared/api/nvmeof.service';
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import { TableComponent } from '~/app/shared/datatable/table/table.component';
+import { CdTableAction } from '~/app/shared/models/cd-table-action';
+import { CdTableColumn } from '~/app/shared/models/cd-table-column';
+import { CdTableFetchDataContext } from '~/app/shared/models/cd-table-fetch-data-context';
+import { CdTableSelection } from '~/app/shared/models/cd-table-selection';
+import { Permission } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { Icons, IconSize } from '~/app/shared/enum/icons.enum';
+import { NvmeofGatewayGroup } from '~/app/shared/models/nvmeof';
+import { CephServiceSpec } from '~/app/shared/models/service.interface';
+
+@Component({
+  selector: 'cd-nvmeof-gateway-group',
+  templateUrl: './nvmeof-gateway-group.component.html',
+  styleUrls: ['./nvmeof-gateway-group.component.scss']
+})
+export class NvmeofGatewayGroupComponent implements OnInit {
+  @ViewChild(TableComponent, { static: true })
+  table: TableComponent;
+
+  @ViewChild('dateTpl', { static: true })
+  dateTpl: TemplateRef<any>;
+
+  @ViewChild('gatewayTpl', { static: true })
+  gatewayTpl: TemplateRef<any>;
+
+  permission: Permission;
+  tableActions: CdTableAction[];
+  columns: CdTableColumn[] = [];
+  selection: CdTableSelection = new CdTableSelection();
+  gatewayGroup$: Observable<CephServiceSpec[]>;
+  private subject = new BehaviorSubject<void>(undefined);
+  context: CdTableFetchDataContext;
+  gatewayGroupName: string;
+  subsystemCount: number;
+  gatewayCount: number;
+
+  icons = Icons;
+
+  iconSize = IconSize;
+
+  constructor(
+    public actionLabels: ActionLabelsI18n,
+    private authStorageService: AuthStorageService,
+    private nvmeofService: NvmeofService
+  ) {}
+
+  ngOnInit(): void {
+    this.permission = this.authStorageService.getPermissions().nvmeof;
+
+    this.columns = [
+      {
+        name: $localize`Name`,
+        prop: 'name'
+      },
+      {
+        name: $localize`Gateways`,
+        prop: 'gatewayCount',
+        cellTemplate: this.gatewayTpl
+      },
+      {
+        name: $localize`Target nodes`,
+        prop: 'nodeCount'
+      },
+      {
+        name: $localize`Subsystems`,
+        prop: 'subSystemCount'
+      },
+      {
+        name: $localize`Created on`,
+        prop: 'created',
+        cellTemplate: this.dateTpl
+      }
+    ];
+
+    this.gatewayGroup$ = this.subject.pipe(
+      switchMap(() =>
+        this.nvmeofService.listGatewayGroups().pipe(
+          switchMap((groupsResponse: GatewayGroup[][]) => {
+            const groups = groupsResponse?.[0] ?? [];
+
+            return forkJoin(
+              groups.map((group: NvmeofGatewayGroup) =>
```

Comment:
why do we have a `GatewayGroup` model and then there is an `NvmeofGatewayGroup` model. What's the difference between them? And why not a single model?